mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1293,28 +1293,78 @@ LogicalResult ModuleImport::convertInstruction(llvm::Instruction *inst) {
-                            unwindArgs);
+    // Check whether the invoke result is an argument to the normal destination
+    // block.
+    std::size_t i = 0;
+    std::optional<std::size_t> invokeBlockArgPos;
+    for (auto inst = invokeInst->getNormalDest()->begin();
+         !invokeBlockArgPos && isa<llvm::PHINode>(inst); ++inst, ++i) {
+      auto *phiInst = cast<llvm::PHINode>(&*inst);
+      llvm::Value *value =
+          phiInst->getIncomingValueForBlock(invokeInst->getParent());
+      if (value == invokeInst)
+        invokeBlockArgPos = i;
+    }
+
+    auto *normalDest = lookupBlock(invokeInst->getNormalDest());
+    auto *directNormalDest = normalDest;
+    if (invokeBlockArgPos) {
+      // The invoke result cannot be an argument to the normal destination
+      // block, as that would imply using the invoke operation result in its
+      // definition, so we need to create a dummy block to serve as an
+      // intermediate destination.
+      OpBuilder::InsertionGuard g(builder);
+      directNormalDest = builder.createBlock(normalDest);
+    }
 
+    SmallVector<Value> unwindArgs;
+    if (failed(convertBranchArgs(invokeInst, invokeInst->getUnwindDest(),
+                                 unwindArgs)))
+      return failure();
+
+    // Create the invoke operation. Normal destination block arguments will be
+    // added later on to handle the case in which the operation result is
+    // included in this list.
     InvokeOp invokeOp;
     if (llvm::Function *callee = invokeInst->getCalledFunction()) {
       invokeOp = builder.create<InvokeOp>(
           loc, types,
           SymbolRefAttr::get(builder.getContext(), callee->getName()), operands,
-          lookupBlock(invokeInst->getNormalDest()), normalArgs,
+          directNormalDest, ValueRange(),
           lookupBlock(invokeInst->getUnwindDest()), unwindArgs);
     } else {
       invokeOp = builder.create<InvokeOp>(
-          loc, types, operands, lookupBlock(invokeInst->getNormalDest()),
-          normalArgs, lookupBlock(invokeInst->getUnwindDest()), unwindArgs);
+          loc, types, operands, directNormalDest, ValueRange(),
+          lookupBlock(invokeInst->getUnwindDest()), unwindArgs);
     }
     if (!invokeInst->getType()->isVoidTy())
       mapValue(inst, invokeOp.getResults().front());
     else
       mapNoResultOp(inst, invokeOp);
+
+    SmallVector<Value> normalArgs;
+    if (failed(convertBranchArgs(invokeInst, invokeInst->getNormalDest(),
+                                 normalArgs)))
+      return failure();
+
+    if (invokeBlockArgPos) {
+      // The dummy normal dest block will just host an unconditional branch
+      // instruction to the normal destination block passing the required block
+      // arguments (including the invoke operation's result).
+      OpBuilder::InsertionGuard g(builder);
+      builder.setInsertionPointToStart(directNormalDest);
+      builder.create<LLVM::BrOp>(loc, invokeOp.getResult(0), normalDest);
```

Comment:
Shouldn't this be

```suggestion
      builder.create<LLVM::BrOp>(loc, normalArgs, normalDest);
```

Looks like you are assuming you can only have 1 argument